apps/examples/connect_test/connect_test_client.c

```diff
@@ -0,0 +1,676 @@
+{
+	char buf[WT_BUF_SIZE];
+	int remain = size;
+	int cnt = 1;
+	int total_sent = 0;
+	CT_LOG(TAG, "start");
+	while (remain > 0) {
+		int send_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+		int nbytes = send(fd, buf, send_size, 0);
+		if (nbytes == 0) {
+			CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+			CT_LOGE(TAG, "connection closed");
+			return -1;
+		} else if (nbytes < 0) {
+			if (errno == EWOULDBLOCK) {
+				CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+				CT_LOGE(TAG, "timeout error %d", errno);
+				return -1;
+			} else {
+				CT_LOG(TAG, "TOTAL BYTES SEND: %d", total_sent);
+				CT_LOGE(TAG, "connection error %d", errno);
+				return -1;
+			}
+		}
+		total_sent += nbytes;
+		remain -= nbytes;
+		cnt++;
+	}
+	CT_LOG(TAG, "end, TOTAL BYTES SEND: %d", total_sent);
+	return 0;
+}
+
+static int _tcp_recv_data(int fd, int size)
+{
+	char buf[WT_BUF_SIZE];
+	int remain = size;
+	int cnt = 1;
+	int total_received = 0;
+	CT_LOG(TAG, "start");
+	while (remain > 0) {
+		int read_size = remain > WT_BUF_SIZE ? WT_BUF_SIZE : remain;
+		int nbytes = recv(fd, buf, read_size, 0);
+		if (nbytes == 0) {
+			CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+			CT_LOGE(TAG, "connection closed");
+			return -1;
+		} else if (nbytes < 0) {
+			if (errno == EWOULDBLOCK) {
+				CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+				CT_LOGE(TAG, "timeout error %d", errno);
+				return -1;
+			} else {
+				CT_LOG(TAG, "TOTAL BYTES RECEIVED: %d", total_received);
+				CT_LOGE(TAG, "connection error %d", errno);
+				return -1;
+			}
+		}
+		total_received += nbytes;
+		remain -= nbytes;
+		cnt++;		
+	}
+	CT_LOG(TAG, "end, TOTAL BYTES RECEIVED: %d", total_received);
+	return 0;
+}
+
+static int _tcp_client(int size)
+{
+	struct sockaddr_in saddr;
+	int ret = 0;
+	int sockfd = 0;
+
+	CT_LOG(TAG, "create socket");
+	int retrycount = 0;
+retry_connection:
+	sockfd = socket(PF_INET, SOCK_STREAM, 0);
+	if (sockfd < 0) {
+		CT_LOGE(TAG, "create socket fail %d", errno);
+		return -1;
+	}
+	
+	/* Set Timeout for Sockets data transfer */
+	struct timeval tv;
+	tv.tv_sec = 30;
+	tv.tv_usec = 0;
+	ret = setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv));
+	int flag = 1;
+	ret = setsockopt(sockfd, IPPROTO_TCP, TCP_NODELAY, (char*) &flag, sizeof(flag));	
+
+	CT_LOG(TAG, "connect to %s:%d", g_tcp_server_address, WT_TCP_PORT);
+
+	/* Connect the socket to the server */
+	saddr.sin_family = AF_INET;
+	saddr.sin_port = htons(WT_TCP_PORT);
+	inet_pton(AF_INET, g_tcp_server_address, &(saddr.sin_addr));
+	unsigned int addrlen = sizeof(struct sockaddr_in);
+	bool is_connected = false;
+	retrycount++;
+	if (connect(sockfd, (struct sockaddr *)&saddr, addrlen) < 0) {
+		CT_LOGE(TAG, "connect fail %d", errno);
+		sleep(1);
+		close(sockfd);
+	} else {
+		CT_LOG(TAG, "connection OK");
+		is_connected = true;
+	}
+
+	if (!is_connected) {
+		if (retrycount <= 5) {
+			CT_LOGE(TAG, "connect try again");
+			goto retry_connection;
```

Comment:
It'll open new socket without closing old socket. In here goto seems not good. because it is difficult to read, and reconnect doesn't requires complicated procedure.

Reply:
Done - Old socket closed before retry connection.